src/tesk_core/job.py

```diff
@@ -17,6 +18,10 @@ def __init__(self, body, name='task-job', namespace='default'):
         self.timeout = 240
         self.body = body
         self.body['metadata']['name'] = self.name
+        self.callback = None
+        if callback_url:
+            task_name = '-'.join(name.split('-')[:2])
```

Comment:
Not quite sure why you process the task name. Is the processed version the one that the user knows (i.e., the one that is returned when `POST`ing a new task)?